lib/Core/TxTree.cpp

```diff
@@ -2252,6 +2352,14 @@ void TxTreeNode::split(ExecutionState *leftData, ExecutionState *rightData) {
   assert(left == 0 && right == 0);
   leftData->txTreeNode = createLeftChild();
   rightData->txTreeNode = createRightChild();
+  if (this->speculationFlag) {
```

Comment:
These lines should be guarded by an if statement which is true when in speculation mode.